common.py

```diff
@@ -400,44 +400,54 @@ def prepare_filelist(self, names, path, goto, indent):
         items += files
         return items
 
-    def is_hidden(self, filename, path, goto=''):
+    def is_hidden(self, filename, path, goto='', get_is_dir=False, dirs_only=False):
         if not (path or goto):  # special case for ThisPC
             return False
         show_hidden = self.show_hidden
         show_excluded = self.view.settings().get('dired_show_excluded_files', True)
         is_hidden = False
-        if not show_hidden:
+        fullpath = join(path, goto, filename)
+        is_dir = isdir(fullpath)
```

Comment:
That is exactly what I call lame in previous comment: we do this check in `traverse_tree` and `prepare_filelist`,
so adding this check here gives overhead (which might be serious in case of many items, [we had two calls of isdir for every name previously](https://github.com/aziz/SublimeFileBrowser/commit/963a023975368b4d2a75c87698cd81af72c6c559)).

Lines 423 is erroneous, because in case of `self.show_hidden is False` the method `is_hidden` must not be called at all, we check it before calling `os.listdir()` in other methods, e.g. `traverse_tree`.

`sublime.platform() == 'windows'` is saved as global variable (constant, if you will) `NT` on top of this file, OS cannot be changed during runtime, so we call it only once on start.

Do not take it as offence or bullying, the issue _is_ complex, I just want it to be bug-free and fast.
Perhaps, it would be better to revert to your initial patch and from there go in direction of a new method which would return two lists (instead one) of directories and files separately, checking if item is hidden before appending it to corresponding list, and ofc adjust depended methods.